core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -379,9 +379,9 @@ public void stopPartitions(Set<TopicPartition> topicPartitions,
                 LOGGER.error("Error while stopping the partition: {}, delete: {}", tpId.topicPartition(), delete, ex);
             }
         });
-        remoteLogMetadataManager.onStopPartitions(topicIdPartitions);
         if (delete) {
             // NOTE: this#stopPartitions method is called when Replica state changes to Offline and ReplicaDeletionStarted
+            remoteLogMetadataManager.onStopPartitions(topicIdPartitions);
```

Comment:
@divijvaidya 

We need to revert this change (see this [comment](https://github.com/apache/kafka/pull/13947/files#r1294768292)).  In the ZK mode, controller sends two stop-partitions request while deleting the topic when the replica state changes to `Offline` and `ReplicaDeletionStarted`.  

RLMM#onStopPartitions causes the TBRLMM consumer to stop consuming records for those respective user-topic-partitions which inturn causes the segment deletion to fail for that topic (since we write a metadata record and read from that topic to denote that the metadata written successfully). 

Stop partition request will be called when a topic is deleted, partition is moved to another replica and node is stopped. In the first 2 cases, the request will have `delete` flag set to true. When the node is stopped, RLMM will be closed anyway.

Reply:
> Stop partition request will be called when a topic is deleted, partition is moved to another replica and node is stopped. 

The 4th case in future is going to be when TS is disabled dynamically for a topic (KIP-950). In such a scenario, TBRLMM will probably still have a problem if the disablement has a "retain=delete". We can solve it in scope for KIP-950 but heads-up that some changes in TBRLMM will be required.

> In the first 2 cases, the request will have delete flag set to true.

When replica is moving to another node, why would `deleteRemoteLog` be true? Yes, the `deleteLocalLog` will be true but not `deleteRemoteLog`. Please help me understand this.

Reply:
yes, you're right. When the replica is moved to another node, then the RLMM#stopPartition won't be called. We can handle this case in TBRLMM by adding a test to ensure that if the same replica is reverted back to the previous node, then it gets handled gracefully. But, we need to handle this case for any RLMM implementation.

Reply:
Addressed your review comments. Please take another look. 

As per the Javadoc in [RemoteLogMetadataManager#onStopPartitions](https://github.com/apache/kafka/blob/trunk/storage/api/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteLogMetadataManager.java#L203), this method will be called when replica emigrates to a new broker and during topic deletion.

> /**
     * This method is invoked only when the topic partitions are stopped on this broker. This can happen when a
     * partition is emigrated to other broker or a partition is deleted.
     *
     * @param partitions topic partitions that have been stopped.
     */
     void onStopPartitions(Set<TopicIdPartition> partitions);

Reply:
Add the respective assertions in the test.